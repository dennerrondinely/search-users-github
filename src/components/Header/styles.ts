import styled, { css } from 'styled-components';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    width: 100%;
    max-width: 60rem;
    margin: 0 auto;
    padding: ${theme.spacings.medium} ${theme.spacings.small} 0;
    display: flex;
  `}
`;
