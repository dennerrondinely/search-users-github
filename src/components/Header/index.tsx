import React, { KeyboardEventHandler, useState } from 'react';

import Icon from 'components/Icon';
import TextField from 'components/TextField';

import * as S from './styles';

export type HeaderProps = {
  onInputUser?: (userLogin: string) => void;
};

//TODO: Encontrar alternativa para keyCode

const Header = ({ onInputUser }: HeaderProps) => {
  const [text, setText] = useState('');
  const handleInputUser: KeyboardEventHandler<HTMLInputElement> = (event) => {
    const { code, currentTarget, keyCode } = event;
    const { value } = currentTarget;
    const valueTrim = value.replace(/( )/g, '');

    console.log(event);

    if ((code === 'Enter' || keyCode === 13) && valueTrim && onInputUser) {
      setText('');
      onInputUser(valueTrim);
    }
  };
  return (
    <S.Wrapper>
      <TextField
        value={text}
        onInputChange={setText}
        icon={<Icon icon="search" />}
        onKeyUp={handleInputUser}
      />
    </S.Wrapper>
  );
};

export default Header;
