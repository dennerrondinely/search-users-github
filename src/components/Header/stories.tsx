import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import Header, { HeaderProps } from '.';

export default {
  title: 'Header',
  component: Header,
  argTypes: {
    onInputUser: { action: 'changed' }
  }
} as Meta;

export const Default: Story<HeaderProps> = (args) => <Header {...args} />;
