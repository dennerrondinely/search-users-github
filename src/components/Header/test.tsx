import React from 'react';
import userEvent from '@testing-library/user-event';
import { fireEvent, render, screen } from 'utils/tests/helpers';

import Header from '.';

describe('<Header />', () => {
  it('should render header', () => {
    const { container } = render(<Header />);

    expect(screen.getByRole('textbox')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should to call onInputUser with typed text', async () => {
    const onInputUser = jest.fn();
    const { container } = render(<Header onInputUser={onInputUser} />);

    const input = screen.getByRole('textbox');
    const text = 'dennerrondinely';
    userEvent.type(input, text);
    fireEvent.keyUp(input, { key: 'Enter', code: 'Enter' });

    expect(onInputUser).toHaveBeenCalledWith(text);

    expect(container.firstChild).toMatchSnapshot();
  });
});
