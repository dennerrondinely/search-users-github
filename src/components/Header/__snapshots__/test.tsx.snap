// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render header 1`] = `
.c4 {
  height: px;
  width: px;
  height: undefinedpx;
  width: undefinedpx;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #EAEDF2;
  border-radius: 1.4rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #EAEDF2;
}

.c5 {
  color: #241911;
  font-family: 'Source Sans Pro','sans-serif','Arial';
  font-size: 1.4rem;
  padding: 0.8rem 0;
  padding-left: 1.6rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c5:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4rem #EAEDF2 inset;
  -webkit-filter: none;
  filter: none;
}

.c5:-webkit-autofill::first-line {
  font-family: 'Source Sans Pro','sans-serif','Arial';
  font-size: 1.6rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #c4c4c4;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c3 > svg {
  width: 1rem;
  height: 100%;
}

.c1 {
  width: 100%;
}

.c0 {
  width: 100%;
  max-width: 60rem;
  margin: 0 auto;
  padding: 3.2rem 2.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          class="c4"
          viewBox="0 0 10 10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.76642 8.58982L7.65286 6.47626C8.22377 5.61371 8.45398 4.56987 8.29886 3.5472C8.14373 2.52453 7.61433 1.59591 6.81332 0.941464C6.01231 0.287014 4.99679 -0.0466256 3.96372 0.00525646C2.93065 0.0571385 1.95366 0.490844 1.22225 1.22225C0.490844 1.95366 0.0571385 2.93065 0.00525646 3.96372C-0.0466256 4.99679 0.287014 6.01231 0.941464 6.81332C1.59591 7.61433 2.52453 8.14373 3.5472 8.29886C4.56987 8.45398 5.61371 8.22377 6.47626 7.65286L8.58982 9.76642C8.74675 9.91799 8.95695 10.0019 9.17512 9.99997C9.3933 9.99807 9.602 9.91056 9.75628 9.75628C9.91056 9.602 9.99807 9.3933 9.99997 9.17512C10.0019 8.95695 9.91799 8.74675 9.76642 8.58982ZM1.68914 4.18547C1.68914 3.69174 1.83555 3.2091 2.10985 2.79858C2.38415 2.38806 2.77402 2.0681 3.23016 1.87916C3.68631 1.69022 4.18824 1.64079 4.67248 1.73711C5.15671 1.83343 5.60152 2.07118 5.95063 2.4203C6.29975 2.76942 6.5375 3.21422 6.63383 3.69846C6.73015 4.1827 6.68071 4.68462 6.49177 5.14077C6.30283 5.59691 5.98287 5.98678 5.57235 6.26109C5.16183 6.53538 4.67919 6.68179 4.18547 6.68179C3.5234 6.68179 2.88845 6.41879 2.4203 5.95063C1.95215 5.48248 1.68914 4.84753 1.68914 4.18547Z"
            stroke-width="0"
          />
        </svg>
      </div>
      <input
        class="c5"
        placeholder="Digite o nome de um usuário..."
        type="text"
        value=""
      />
    </div>
  </div>
</header>
`;

exports[`<Header /> should to call onInputUser with typed text 1`] = `
.c4 {
  height: px;
  width: px;
  height: undefinedpx;
  width: undefinedpx;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #EAEDF2;
  border-radius: 1.4rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #EAEDF2;
}

.c5 {
  color: #241911;
  font-family: 'Source Sans Pro','sans-serif','Arial';
  font-size: 1.4rem;
  padding: 0.8rem 0;
  padding-left: 1.6rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c5:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4rem #EAEDF2 inset;
  -webkit-filter: none;
  filter: none;
}

.c5:-webkit-autofill::first-line {
  font-family: 'Source Sans Pro','sans-serif','Arial';
  font-size: 1.6rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #c4c4c4;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c3 > svg {
  width: 1rem;
  height: 100%;
}

.c1 {
  width: 100%;
}

.c0 {
  width: 100%;
  max-width: 60rem;
  margin: 0 auto;
  padding: 3.2rem 2.4rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          class="c4"
          viewBox="0 0 10 10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.76642 8.58982L7.65286 6.47626C8.22377 5.61371 8.45398 4.56987 8.29886 3.5472C8.14373 2.52453 7.61433 1.59591 6.81332 0.941464C6.01231 0.287014 4.99679 -0.0466256 3.96372 0.00525646C2.93065 0.0571385 1.95366 0.490844 1.22225 1.22225C0.490844 1.95366 0.0571385 2.93065 0.00525646 3.96372C-0.0466256 4.99679 0.287014 6.01231 0.941464 6.81332C1.59591 7.61433 2.52453 8.14373 3.5472 8.29886C4.56987 8.45398 5.61371 8.22377 6.47626 7.65286L8.58982 9.76642C8.74675 9.91799 8.95695 10.0019 9.17512 9.99997C9.3933 9.99807 9.602 9.91056 9.75628 9.75628C9.91056 9.602 9.99807 9.3933 9.99997 9.17512C10.0019 8.95695 9.91799 8.74675 9.76642 8.58982ZM1.68914 4.18547C1.68914 3.69174 1.83555 3.2091 2.10985 2.79858C2.38415 2.38806 2.77402 2.0681 3.23016 1.87916C3.68631 1.69022 4.18824 1.64079 4.67248 1.73711C5.15671 1.83343 5.60152 2.07118 5.95063 2.4203C6.29975 2.76942 6.5375 3.21422 6.63383 3.69846C6.73015 4.1827 6.68071 4.68462 6.49177 5.14077C6.30283 5.59691 5.98287 5.98678 5.57235 6.26109C5.16183 6.53538 4.67919 6.68179 4.18547 6.68179C3.5234 6.68179 2.88845 6.41879 2.4203 5.95063C1.95215 5.48248 1.68914 4.84753 1.68914 4.18547Z"
            stroke-width="0"
          />
        </svg>
      </div>
      <input
        class="c5"
        placeholder="Digite o nome de um usuário..."
        type="text"
        value=""
      />
    </div>
  </div>
</header>
`;
